packages/svelte/src/internal/server/types.d.ts

```diff
@@ -1,21 +1,33 @@
-export interface Component {
+import type { Element } from './dev';
+import type { Payload } from './payload';
+
+export interface SSRContext {
 	/** parent */
-	p: null | Component;
-	/** context */
+	p: null | SSRContext;
+	/** component context */
 	c: null | Map<unknown, unknown>;
-	/** ondestroy */
-	d: null | Array<() => void>;
-	/**
-	 * dev mode only: the component function
-	 */
+	/** payload (renderer) */
+	r: null | Payload;
+	/** dev mode only: the current component function */
 	function?: any;
+	/** dev mode only: the current element */
+	element?: Element;
 }
 
-export interface RenderOutput {
+export interface SyncRenderOutput {
 	/** HTML that goes into the `<head>` */
 	head: string;
 	/** @deprecated use `body` instead */
 	html: string;
 	/** HTML that goes somewhere into the `<body>` */
 	body: string;
 }
+
+export type RenderOutput = SyncRenderOutput & Thenable<SyncRenderOutput>;
+
+interface Thenable<T> {
+	then<TResult1 = T, TResult2 = never>(
+		onfulfilled?: ((value: T) => TResult1 | Thenable<TResult1>) | undefined | null,
+		onrejected?: ((reason: any) => TResult2 | Thenable<TResult2>) | undefined | null
```

Comment:
come to think of it shouldn't it be this? I guess you can return any valid `Promise.resolve(...)` argument but it feels weird that we'd have to supply that type

```suggestion
		onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null,
		onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null
```

Reply:
Where are you getting `PromiseLike` from?

Reply:
it's a built-in

Reply:
`Thenable` appears to be a clone of the `PromiseLike` built-in.

https://github.com/microsoft/TypeScript/blob/e9bcbe6ef706e0b5a34678964988eb6a9cd86cc6/src/lib/es5.d.ts#L1519